extensions/amp-youtube/1.0/youtube.js

```diff
@@ -0,0 +1,227 @@
+      return;
+    }
+    if (data.event == 'initialDelivery') {
+      dispatchCustomEvent(currentTarget, VideoEvents.LOADEDMETADATA);
+      return;
+    }
+    const {info} = data;
+    if (info == undefined) {
+      return;
+    }
+    const playerState = info['playerState'];
+    if (data.event == 'infoDelivery' && playerState == 0 && loop) {
+      currentTarget.contentWindow./*OK*/ postMessage(
+        JSON.stringify(
+          dict({
+            'event': 'command',
+            'func': 'playVideo',
+          })
+        ),
+        '*'
+      );
+    }
+    if (data.event == 'infoDelivery' && playerState != undefined) {
+      dispatchCustomEvent(currentTarget, PlayerStates[playerState.toString()]);
+    }
+    if (data.event == 'infoDelivery' && info['muted']) {
+      dispatchCustomEvent(currentTarget, mutedOrUnmutedEvent(info['muted']));
+      return;
+    }
+  };
+
+  const sendMessage = (method) => {
+    return JSON.stringify(
+      dict({
+        'event': 'command',
+        'func': methods[method],
+      })
+    );
+  };
+
+  return (
+    <VideoWrapper
+      component={VideoIframe}
+      autoplay={autoplay}
+      src={src}
+      style={style}
+      size={true}
+      layout={true}
+      paint={true}
+      onMessage={onMessage}
+      sendMessage={sendMessage}
+      onLoad={(event) => {
+        const {currentTarget} = event;
+        dispatchCustomEvent(currentTarget, 'canplay');
+        currentTarget.contentWindow./*OK*/ postMessage(
+          JSON.stringify(
+            dict({
+              'event': 'listening',
+            })
+          ),
+          '*'
+        );
+        dispatchCustomEvent(currentTarget, VideoEvents.LOAD);
+      }}
+      sandbox="allow-scripts allow-same-origin allow-presentation"
+      {...rest}
+    ></VideoWrapper>
+  );
+}
+
+/**
+ * @param {string} credentials
+ * @param {string} videoid
+ * @param {string} liveChannelid
+ * @return {string}
+ * @private
+ */
+function getEmbedUrl(credentials, videoid, liveChannelid) {
+  let urlSuffix = '';
+  if (credentials === 'omit') {
+    urlSuffix = '-nocookie';
+  }
+  const baseUrl = `https://www.youtube${urlSuffix}.com/embed/`;
+  let descriptor = '';
+  if (videoid) {
+    descriptor = `${encodeURIComponent(videoid)}?`;
+  } else {
+    descriptor = `live_stream?channel=${encodeURIComponent(
+      liveChannelid || ''
+    )}&`;
+  }
+  return `${baseUrl}${descriptor}enablejsapi=1&amp=1`;
+}
+
+/**
+ * Dispatches a custom event.
+ *
+ * @param {HTMLIFrameElement} currentTarget
+ * @param {string} name
+ * @param {!Object=} opt_data Event data.
+ * @final
+ */
+function dispatchCustomEvent(currentTarget, name, opt_data) {
+  const data = opt_data || {};
+  // Constructors of events need to come from the correct window. Sigh.
+  const event = currentTarget.ownerDocument.createEvent('Event');
+  event.data = data;
+  event.initEvent(name, /* bubbles */ true, /* cancelable */ true);
+  currentTarget.dispatchEvent(event);
+}
```

Comment:
Use `createCustomEvent` from `src/event-helper`

```suggestion
function dispatchCustomEvent(currentTarget, name, opt_data) {
  // Constructors of events need to come from the correct window. Sigh.
  const event = createCustomEvent(currentTarget.ownerDocument, name, opt_data, {
    bubbles: true,
    cancelable: true,
  });
  currentTarget.dispatchEvent(event);
}
```